libs/SalesforceSDK/src/com/salesforce/androidsdk/rest/RestRequest.java

```diff
@@ -709,6 +714,103 @@ public static RestRequest getRequestForSObjectTree(String apiVersion, String obj
+     */
+    public static RestRequest getRequestForNotificationUpdate(String apiVersion, String notificationId, Boolean read, Boolean seen) {
+        final Map<String, Object> parameters = new HashMap<>();
+        if (read != null) {
+            parameters.put("read", read);
+        }
+        if (seen != null) {
+            parameters.put("seen", seen);
+        }
+        final String path = RestAction.NOTIFICATIONS.getPath(apiVersion, notificationId);
+        return new RestRequest(RestMethod.PATCH, path, new JSONObject(parameters));
+    }
+
+    /**
+     * Request for getting notifications.
+     *
+     * @param apiVersion   Salesforce API version.
+     * @param size         Number of notifications to get.
+     * @param before       Get notifications occurring before the provided date. Shouldn't be used with `after`.
+     * @param after        Get notifications occurring after the provided date. Shouldn't be used with `before`.
+     */
+    public static RestRequest getRequestForNotifications(String apiVersion, Integer size, Date before, Date after) {
+        final Map<String, String> parameters = new HashMap<>();
+        if (size != null) {
+            parameters.put("size", size.toString());
+        }
+        if (before != null) {
+            parameters.put("before", ISO8601_DATE_FORMAT.format(before));
+        }
+        if (after != null) {
+            parameters.put("after", ISO8601_DATE_FORMAT.format(after));
+        }
+
+        final ConnectUriBuilder builder = new ConnectUriBuilder(Uri.parse(RestAction.NOTIFICATIONS.getPath(apiVersion, "")).buildUpon());
+        for (final Map.Entry<String, String> parameter : parameters.entrySet()) {
+            builder.appendQueryParam(parameter.getKey(), parameter.getValue());
+        }
+        return new RestRequest(RestMethod.GET, builder.toString());
+    }
+
+    /**
+     * Request for updating notifications.
+     *
+     * @param apiVersion       Salesforce API version.
+     * @param notificationIds  IDs of notifications to get. Shouldn't be used with `before`.
+     * @param before           Get notifications before the provided date. Shouldn't be used with `notificationIds`.
+     * @param read             Marks notifications as read (true) or unread (false). If null, field won't be updated.
+     *                         Required if `seen` not provided.
+     * @param seen             Marks notifications as seen (true) or unseen (false). If null, field won't be updated.
+     *                         Required if `read` not provided.
+     */
+    public static RestRequest getRequestForNotificationsUpdate(String apiVersion, List<String> notificationIds, Date before, Boolean read, Boolean seen) {
+        final Map<String, Object> parameters = new HashMap<>();
+        if (notificationIds != null) {
+            parameters.put("notificationIds", notificationIds);
+        }
+        if (before != null) {
```

Comment:
Should we check that they haven't supplied both `notificationIds` and `before`?  Or does the API handle that?

Reply:
Nah - I think we rely on the doc for that (both ours and the server side doc). If the user supplies both, the endpoint will throw an error.